### PR TITLE
feat(sidecar): gross tip revenue metric

### LINKED ISSUE #245: Sidecar metrics for revenue
## Context
Proposers should have a way to track their revenue (in priority fees) from the commitments they make. 
Note that this should only be incremented when a preconf is actually included in their block.

The calculation should be the *effective* tip per gas x the gas the transaction actually consumed. Effective tip can be calculated with `tx.effective_tip_per_gas(basefee)` which requires the basefee of the block.

### PR DESCRIPTION
**Pull request by mempirate**
Closes #245

### CODE REVIEW

File: .github/workflows/bolt_sidecar_ci.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ name: Bolt Sidecar CI
 
 on:
   push:
+    branches:
+      - unstable
+      - main
```
Comment (thedevbirb) [inline .github/workflows/bolt_sidecar_ci.yml:7]: The `main` branch is unused. Any specific reason for adding it?

File: bolt-sidecar/src/telemetry/metrics.rs [hunk 2]
```diff
@@ -81,6 +86,25 @@ impl ApiMetrics {
...
+            }
+
+            parts.push(tip as u64);
+
+            for part in parts {
+                counter!(GROSS_TIP_REVENUE).increment(part);
+            }
+        } else {
+            counter!(GROSS_TIP_REVENUE).increment(tip as u64);
+        }
```
Comment (thedevbirb) [inline bolt-sidecar/src/telemetry/metrics.rs:105]: In my opinion this is a bit simpler/succint and doesn't distinguish between edge cases and regular cases which is nice.
```suggestion
         // Takes also into account tips larger than a u128 by splitting them into u64 parts
        let mut parts = Vec::new();
        let u64_max = u64::MAX as u128;

        while tip > 0 {
            // We can only add a max tip of u64::MAX.
            let min = tip.min(u64_max) as u64;
            parts.push(min);
            // Subtract the part we just added, if something is left then tip is going to be
            // greater than zero.
            tip = tip.saturating_sub(u64_max)
        }

        parts.push(tip as u64);

        for part in parts {
            counter!(GROSS_TIP_REVENUE).increment(part);
        }
```

**Review by thedevbirb — COMMENTED** [review]
I've not seen the metric added anywhere in the sidecar code, is this PR still draft?

File: .github/workflows/bolt_sidecar_ci.yml (hunk 1 above)
Comment (mempirate) [reply .github/workflows/bolt_sidecar_ci.yml:7]: We might use it in the future

File: bolt-sidecar/src/telemetry/metrics.rs (hunk 2 above)
Comment (mempirate) [reply bolt-sidecar/src/telemetry/metrics.rs:105]: True, but this always does an allocation

**Comment by mempirate** [comment]
> I've not seen the metric added anywhere in the sidecar code, is this PR still draft?

Yes mb, converted

File: bolt-sidecar/src/telemetry/metrics.rs (hunk 2 above)
Comment (thedevbirb) [reply bolt-sidecar/src/telemetry/metrics.rs:105]: That's fair, also because with an u64 you can express a tip up to 18.44 ETH :D

**Review by merklefruit — APPROVED** [review]

**Review by thedevbirb — APPROVED** [review]
Great!

**Pull request merged**